### PR TITLE
Updated the ddl2cpp script to allow custom/extended types

### PR DESCRIPTION
**Pull request by CJCombrink**
…ernal script

- Had to re-order the script to allow the command line to be parsed before setting up the parser
- Added an example python file to show usage
- Fixed tests as well

Implements #485

### CODE REVIEW

File: scripts/ddl2cpp [hunk 1]
```diff
@@ -485,6 +494,19 @@ def setArgumentBool(s, bool_value):
     var_name = first_lower(re.sub("(\s|-|[0-9])(\S)", repl_func_for_args, s))
     globals()[var_name] = bool_value
 
+def loadExtendedTypesFile(filename):
```
Comment (CJCombrink) [inline scripts/ddl2cpp:497]: One can probably make an argument for not loading a python script but a csv or other text file. I am not very proficient with python and decided with going this route. It should be easy enough to update/extend at this function without having to touch the rest of the code again.

Comment (rbock) [reply scripts/ddl2cpp:497]: Could you try to go the CSV route? Reading CSV is quite simple in python.

**Review by rbock — CHANGES_REQUESTED** [review]
Thanks for creating this PR!

The commit message starts with `...`? That looks a bit like an accident?

Did you forget to add the tests you mentioned? There should be be at least one test using the new command line switch.

File: scripts/ddl2cpp (hunk 1 above)
Comment (CJCombrink) [reply scripts/ddl2cpp:497]: Sure I can give it a bash and see what I get

**Comment by CJCombrink** [comment]
> Thanks for creating this PR!
> 
> The commit message starts with `...`? That looks a bit like an accident?
> 
> Did you forget to add the tests you mentioned? There should be be at least one test using the new command line switch.

File: datatypes.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# This file can be copied, updated and then passed to ddl2cpp as a command line argument
+# Each variable in this file must be a list of case insensitive type names.
+# Usage:
+#    ddl2cpp [other optional args] --datatype-file=/path/to/datatypes.py <path to ddl> <path to target> <namespace>
+
+extendedDdlBooleanTypes = []
+extendedDdlIntegerTypes = []
+extendedDdlSerialTypes = []
+extendedDdlFloatingPointTypes = []
+extendedDdlTextTypes = ["UUID", "other_type"]
```
[the remark this answers is not in the document]
Comment (CJCombrink) [reply datatypes.py:12]: Was thinking more of an example

File: scripts/ddl2cpp (hunk 1 above)
Comment (CJCombrink) [reply scripts/ddl2cpp:497]: Do you have an idea or suggestion on how a CSV would look for this?
Maybe:
```
extendedDdlIntegerTypes, "one", "two"
```
I think this is what tripped me up when I considered the approach, I could not think of a nice other format

Comment (CJCombrink) [reply scripts/ddl2cpp:497]: I think I have something that can work, will create a new PR once it works

**Pull request closed without merging**